### ISSUE TITLE
still Device not found. Ensure your device is powered on and connected to your network.

### ISSUE THREAD

**Issue by rubyon** [jkaberg/tvhProxy#21]
Channel setting OK.
But Not working with this message "Device not found. Ensure your device is powered on and connected to your network."

My PMS Version 1.7.2.3878-8088811b8

TVH Version 4.3-32~g696e17d51

**Comment by regnets** [comment]
Same here with TVH Verison 4.09.

**Comment by matbor** [comment]
@rubyon @regnets see if this works, https://github.com/jkaberg/tvhProxy/issues/15#issuecomment-306679456

**Comment by Tezzlicious** [comment]
Same problem here. The solution doesn't work anymore. Worked on PMS 1.5x.

**Comment by Marlinc** [comment]
I'm running into the same problem at the moment

**Comment by NoMoreNicksLeft** [comment]
I'm not using this software, but similar. Same problem there. If you look in the PMS Server logs, you'll notice that recent Plex versions now request a new xml file from the device, "capability".

I have na actual HDHomerun though, and it does not have this file on it at all. I can't find any references to the file, and I can't guess what's supposed to be inside of it (except that it's xml... if you serve up a blank file, or garbage, the Plex logs gripe that it's not valid xml).

Presumably if tvhproxy served up this file, things would work again (and so would my software).

I'm wondering if it's not a response to all the Kodi streaming bullshit that happened earlier this year. Plex Inc. might be trying to limit the functionality to only genuine devices to avoid legal repercussions. If anyone knows anything, would love to hear about it.

**Comment by rndthoughts** [comment]
It is possible to disable this check by editing Preferences,xml (see https://support.plex.tv/hc/en-us/articles/201105343-Advanced-Hidden-Server-Settings) and setting disableCapabilityCheck="1" in the xml file.   This removes the message.

**Comment by irishj** [comment]
Same issue here, it was working on previous Plex Server version

Currently running Plex Server v1.8.3.4235
Currently running TVH v4.2.3-39

**Comment by si458** [comment]
same issue here

Plex: V1.8.4.4249
TVH: 4.2.3 (2017-07-01T08:15:26+0000)
tvhProxy: latest from git master

**Comment by im85288** [comment]
Hi all, I made a fix for this and created a pull request here: https://github.com/jkaberg/tvhProxy/pull/31

You can either wait for the pull request to be merged, copy the changes from the PR or try from my fork https://github.com/im85288/tvhProxy

**Comment by si458** [comment]
@im85288 it worked for about 15mins then stopped, every time it says its offline i have to restart plexmediaserver and it then says back online

also device.xml only gives me the template output and im sure it should have info in it?
`<root><specVersion><major>1</major><minor>0</minor></specVersion><URLBase/><device><deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType><friendlyName/><manufacturer/><modelName/><modelNumber/><serialNumber/><UDN>uuid:</UDN></device></root>`

**Comment by si458** [comment]
@im85288 i have made a pull request on ur repo, the discover function was returning a Repsonse variable and not a JSON response so the XML template couldnt understand it and was missing values

**Comment by im85288** [comment]
@si458 thanks I did notice that but for me it worked regardless of having those values populated. I will though pull in your changes.

**Comment by irishj** [comment]
Same issue, even after moving to your fork and setting up again.  Worked initally, setup dvr and pulled in EPG, but going back into dvr settings today still shows "Device not found. Ensure your device is powered on and connected to your network".

**Comment by si458** [comment]
@irishj if you restart the plexmediaserver, does it show up as online? Because from what I've seen/debugged the plexmediaserver is caching the dvr stuff which seems odd?

**Comment by irishj** [comment]
Just re-started plex server and yes, it now shows up as being online again.  Very odd ?!?

**Comment by im85288** [comment]
No idea what’s happening as it’s fine on my setup (Ubuntu Linux flavour of Plex latest Plex Pass version)

**Comment by im85288** [comment]
Well upgraded to version 1.9.0.4252 of the plex server and it no longer shows up at all...anybody know what has changed or how to debug what plex is calling...

**Comment by im85288** [comment]
Hmm then after a restart of the server all is good again..bizarre but seems to work

**Comment by si458** [comment]
@im85288 my plex auto upgrades every night to latest plex pass version (https://github.com/mrworf/plexupdate) and no issues with mine so far?

**Comment by im85288** [comment]
Thanks @si458 I hadn’t seen that script, I use LibreElec though and follow this guide to manually update: https://www.naschenweng.info/2016/12/23/upgrade-plex-media-server-plexpass-libreelec-command-line/ I only had the issue after the initial install of the new version. So your caching theory seems to make sense. Cheers

**Comment by si458** [comment]
only reason i think the is a caching issue is because i dont see requests to the tvhProxy unless u are in the Live DVR tab in settings but also last night while recording a stream my iptv provider dropped but plex carried on recording but after i restarted tvheadend plex was saying it was still recording which is strange?

**Comment by im85288** [comment]
Yeah I had a failed recording from my HDHomeRun device yesterday, in fact rather last few recordings failed. I had to reset the HDHomeRun device to free the tuner in Plex. I’ll try and record an iptv channel to test as that worked fine in the past, though of course that depends on the stream not dropping
